### PR TITLE
docs: update core concepts sections

### PR DESCRIPTION
**Pull request by dessaya**

### CODE REVIEW

File: documentation/docs/contribute.md [hunk 1]
```diff
@@ -28,26 +28,16 @@ Before creating a pull request ensure that all tests pass locally, and that the
...
-::: warn
-
-Check that the `database.inMemory` parameter is set to `false` (default) before running the complete test suite. See the [source code](https://github.com/iotaledger/wasp/blob/develop/tools/cluster/tests/spam_test.go) for details.
-
-:::
-
-```shell
-make test-full
-```
+The commands above execute a subset of all available tests. If you introduced major changes, consider running the whole test suite instead, with `make test` or `make test-full` (these can take several minutes, so go and grab a coffee!).
```
Comment (jorgemmsilva) [inline documentation/docs/contribute.md:40]: :coffee:  :smile:

File: documentation/docs/guide/core_concepts/consensus.md [hunk 2]
```diff
@@ -25,10 +26,12 @@ Then the nodes have to agree on which batch proposals they want to work on. In s
 
 The next step is to convert the raw list of batch proposals into an actual batch: for that, all requests from all proposals are counted and filtered to produce the same single list of requests in the same order.
 
-As all nodes had to sign the same piece of information, the hash of the current state, these partial signatures can be combined into a full, valid signature that is then fed to a pseudo-random function that orders the smart contract requests. Validator nodes can neither affect nor predict the final order of requests in the batch.
+As all nodes had to sign the same piece of information: the hash of the current state.
```
Comment (jorgemmsilva) [inline documentation/docs/guide/core_concepts/consensus.md]: nitpick: I would say "a cryptographic commitment to the current state", its more that just an hash

File: documentation/docs/guide/core_concepts/consensus.md [hunk 3]
```diff
@@ -25,10 +26,12 @@ Then the nodes have to agree on which batch proposals they want to work on. In s
...
+These partial signatures can be combined into a full, valid signature that is then fed to a pseudo-random function that sorts the smart contract requests.
+Validator nodes can neither affect nor predict the final order of requests in the batch.
```
Comment (jorgemmsilva) [inline documentation/docs/guide/core_concepts/consensus.md]: suggestion: we could mention here that this protects ISC from MEV attacks

File: documentation/docs/guide/core_concepts/invocation.md [hunk 4]
```diff
@@ -15,22 +15,35 @@ keywords:
...
+Being able to combine smart contracts in this way is called *synchronous composability*.
 
 ## Requests
 
-A request contains a call to a smart contract and a signature of the sender (who is also the owner of the assets and funds that are going to be processed within the request). Unlike calls, requests are not executed immediately. Instead, they have to wait until the chain's validator nodes include them into a request batch. This means that requests unlike the calls have a delay and are executed in an unpredictable order, but they are subject to the protection mechanisms that allow ISC to function.
+A request contains a call to a smart contract and a signature of the sender (who is also the owner of the assets and funds that are going to be processed within the request).
+Unlike calls between smart contracts, requests are not executed immediately.
+Instead, they have to wait until the chain's validator nodes include them into a request batch.
+This means that requests have a delay and are executed in an unpredictable order (but they are subject to the protection mechanisms that allow ISC to function).
```
Comment (jorgemmsilva) [inline documentation/docs/guide/core_concepts/invocation.md]: >(but they are subject to the protection mechanisms that allow ISC to function).
what does this mean? - I know its existing text, but I would remove it

File: documentation/docs/guide/core_concepts/validators.md [hunk 5]
```diff
@@ -12,4 +13,24 @@ keywords:
...
+## Access nodes
+
+It is possible to have some nodes that act as _access nodes_ to the chain, without being be part of the committee of validators.
+All nodes in the subnet (validators and non-validators) are connected to each other through statically assigned trust relationships, and each node is also connected to the IOTA L1 node in order to receive updates on the chain’s L1 account.
+
+Any node can optionally provide access to smart contracts for external callers, allowing them to:
+* Query the state of the chain (i.e. _view calls_)
+* Send off-ledger requests directly to the node (instead of sending an on-ledger request as a L1 transaction).
+
+It is common for validator nodes to be part of a private subnet, and have only a group of access nodes exposed to the outside world, protecting the committee from external attacks.
```
Comment (jorgemmsilva) [inline documentation/docs/guide/core_concepts/validators.md:35]: I guess we should add some description of how the management of access nodes is done via the governance contract

File: documentation/docs/overview.md [hunk 6]
```diff
@@ -13,32 +13,42 @@ keywords:
...
+This allows parties that need private blockchains to use the security of the public IOTA network without actually exposing their data to the public.
 
 ### Flexibility
 
-You can run multiple types of virtual machines on any chain. We are starting with 
-[Rust/Wasm](https://rustwasm.github.io/docs/book/)-based smart contracts, followed by 
-[Solidity/EVM](https://docs.soliditylang.org/en/v0.8.6/)-based smart contracts, but eventually all kinds of virtual machines can be supported in a IOTA Smart Contract chain depending on the demand. 
+ISC is agnostic about the virtual machine that actually executes the smart contract code.
+Currently we support [Rust/Wasm](https://rustwasm.github.io/docs/book/)-based smart contracts and [Solidity/EVM](https://docs.soliditylang.org/en/v0.8.6/)-based smart contracts.
+Eventually all kinds of virtual machines can be supported in an ISC chain depending on the demand.
```
Comment (jorgemmsilva) [inline documentation/docs/overview.md:42]: I bet JVM would be popular :trollface:

**Review by jorgemmsilva — APPROVED** [review]

File: documentation/docs/guide/core_concepts/states.md [hunk 7]
```diff
@@ -16,74 +16,83 @@ keywords:
...
+We call this output the *state output* (or state anchor), and the containing transaction the *state transaction* (or anchor transaction) of the chain.
 The state output is controlled (i.e. can be unlocked/consumed) by the entity running the chain.
 
-With the anchoring mechanism the UTXO ledger supports the IOTA Smart Contracts chain the following ways:
+With the anchoring mechanism, the UTXO ledger provides the following guarantees to the IOTA Smart Contracts chain:
 
-- Guarantees global consensus on the state of the chain
-- Makes the state immutable and tamper-proof
-- Makes the state consistent (see below)
+- There is a global consensus on the state of the chain.
```
Comment (lmoe) [inline documentation/docs/guide/core_concepts/states.md]: nit: I would write lists without dots at the end

**Review by lmoe — APPROVED** [review]
Reads great :)

**Pull request merged**